test/session_replay_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:instabug_flutter/instabug_flutter.dart';
+import 'package:instabug_flutter/src/generated/session_replay.api.g.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'session_replay_test.mocks.dart';
+
+@GenerateMocks([
+  SessionReplayHostApi,
+])
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  WidgetsFlutterBinding.ensureInitialized();
+
+  final mHost = MockSessionReplayHostApi();
+
+  setUpAll(() {
+    SessionReplay.$setHostApi(mHost);
+  });
+
+  test('[enabled] should call host method', () async {
```

Comment:
Let's keep the test names in sync by using the method name here.

```suggestion
  test('[setEnabled] should call host method', () async {
```